tests/test-coauthors-plus.php

```diff
@@ -679,6 +699,1434 @@ public function test_update_author_term_when_author_term_not_exist() {
+	 *
+	 * @return void
+	 */
+	public function test_append_post_author_who_has_not_been_linked(  ) {
+		$post_id = $this->factory()->post->create(
+			array(
+				'post_author' => $this->author2->ID,
+				'post_status' => 'publish',
+				'post_type'   => 'post',
+			)
+		);
+
+		$this->_cap->add_coauthors( $post_id, array( $this->author3->user_login ), true );
+
+		$query = new WP_Query(
+			array(
+				'p' => $post_id,
+			)
+		);
+
+		$this->assertEquals( 1, $query->found_posts );
+		$this->assertEquals( $this->author2->ID, $query->posts[0]->post_author );
+
+		$post_author_terms = wp_get_post_terms( $post_id, $this->_cap->coauthor_taxonomy );
+
+		$this->assertTrue( is_array( $post_author_terms ) );
+		$this->assertCount( 2, $post_author_terms );
+
+		$author_slugs = array(
+			'cap-' . $this->author2->user_login,
+			'cap-' . $this->author3->user_login,
+		);
+
+		foreach ( $post_author_terms as $term ) {
+			$this->assertInstanceOf( WP_Term::class, $term );
+			$this->assertContains( $term->slug, $author_slugs );
+		}
+	}
+
+	/**
+	 * Here we are assigning multiple authors who have not been
+	 * linked to a coauthor to a post. Since we are not
+	 * appending authors to the post, we should
+	 * expect the post_author to change.
+	 *
+	 * @return void
+	 */
+	public function test_assign_post_authors_from_authors_who_have_not_been_linked() {
+		$post_id = $this->factory()->post->create(
+			array(
+				'post_author' => $this->author1->ID,
+				'post_status' => 'publish',
+				'post_type'   => 'post',
+			)
+		);
+
+		$this->_cap->add_coauthors(
+			$post_id,
+			array(
+				$this->author3->user_login,
+				$this->editor1->user_login,
+				$this->author2->user_login,
+			)
+		);
+
+		$query = new WP_Query(
+			array(
+				'p' => $post_id,
+			)
+		);
+
+		$this->assertEquals( 1, $query->found_posts );
+		$this->assertEquals( $this->author3->ID, $query->posts[0]->post_author );
+
+		$post_author_terms = wp_get_post_terms( $post_id, $this->_cap->coauthor_taxonomy );
+
+		$this->assertTrue( is_array( $post_author_terms ) );
+		$this->assertCount( 3, $post_author_terms );
+
+		$author_slugs = array(
+			'cap-' . $this->author3->user_login,
+			'cap-' . $this->editor1->user_login,
+			'cap-' . $this->author2->user_login,
+		);
+
+		foreach ( $post_author_terms as $term ) {
+			$this->assertInstanceOf( WP_Term::class, $term );
+			$this->assertContains( $term->slug, $author_slugs );
+		}
+	}
+
+	/**
+	 * Here we are creating guest authors (coauthors) and assigning them to a post,
+	 * which was created by a WP_User. Since the guest authors have not been
+	 * linked to a WP_User, the wp_post.post_author column should not
+	 * change, and the response from CoAuthors_Plus::add_coauthors()
+	 * should be false, since no WP_User could be found.
+	 *
+	 * @return void
+	 */
+	public function test_assign_post_authors_from_coauthors_who_have_not_been_linked() {
+		$random_username = 'random_user_' . rand( 1, 1000 );
```

Comment:
Could use [WP_UnitTest_Generator_Sequence](https://core.trac.wordpress.org/browser/trunk/tests/phpunit/includes/factory/class-wp-unittest-generator-sequence.php) here so that the next random username doesn't need to be between 1001 and 2000, etc.